File: 3-Patron/php/Mazmorra.php
<?php
class Mazmorra{
	private $salas = [];

	public function __construct($salas){
    $this->salas = $salas;
	}
	public function __construct(){}
	
	public function getSalas(){
		return $this->salas;	
	}

	public function setSalas($salas){
		$this->salas = $salas;
	}

	public function addSala($sala){
		$this->salas = $salas;
		array_push($this->salas,$sala);
	}
}
?>

File: 3-Patron/Java/Mazmorra.java
package buildercomposite;
import java.util.*;

public class Mazmorra {
    private ArrayList<IMapa> Salas = new ArrayList<IMapa>();

    public Mazmorra(ArrayList<IMapa> salas){
        this.Salas = salas;
    }

    public Mazmorra() {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }
    
    public ArrayList<IMapa> getSalas(){
        return this.Salas;
    }
    
    public void setSalas(ArrayList<IMapa> salas){
        this.Salas = salas;
    }
    public void addSala(IMapa sala){
        this.Salas.add(sala);
    }
}
